the-blue-alliance-ios/View Controllers/Match/MatchQueryOptionsViewController.swift

```diff
@@ -0,0 +1,126 @@
+import CoreData
+import Foundation
+import TBAKit
+import UIKit
+
+protocol MatchQueryOptionsDelegate: AnyObject {
+    func updateQuery(query: MatchQueryOptions)
+}
+
+// Backing enums to setup our table view data source
+private enum QuerySections: String, CaseIterable {
+    case sort = "Sort"
+    case filter = "Filter"
+}
+
+private enum SortRows: String, CaseIterable {
+    case reverse = "kMatchSortReverse"
+}
+
+private enum FilterRows: String, CaseIterable {
+    case favorites = "kMatchFilterFavorites"
+}
+
+// Backing structs to power our data
+struct MatchQueryOptions {
+    var sort: MatchSortOptions
+    var filter: MatchFilterOptions
+
+    struct MatchSortOptions {
+        var reverse: Bool
+    }
+
+    struct MatchFilterOptions {
+        var favorites: Bool
+    }
+}
+
+class MatchQueryOptionsViewController: TBATableViewController {
+
+    private var query: MatchQueryOptions
+
+    weak var delegate: MatchQueryOptionsDelegate?
+
+    init(persistentContainer: NSPersistentContainer, tbaKit: TBAKit, userDefaults: UserDefaults) {
+        query = MatchQueryOptions(sort: MatchQueryOptions.MatchSortOptions(reverse: false), filter: MatchQueryOptions.MatchFilterOptions(favorites: false))
+        super.init(style: .plain, persistentContainer: persistentContainer, tbaKit: tbaKit, userDefaults: userDefaults)
+
+        title = "Match Sort/Filter"
+        
+        navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .done, target: self, action: #selector(saveMatchQuery))
```

Comment:
This needs to get updated to the new system of setting right bar button items

Reply:
jk - we're using `TBATableViewController` in a weird way where we need to set this directly